### ISSUE TITLE
{no_ack, false} not implemented

### ISSUE THREAD

**Issue by ypaq** [muxspace/bunny_farm#15]
Hey Brian,

I was looking through the code and couldn't find a way to explicitly acknowledge messages. Am I missing something or is there no way implemented to ack messages with gen_qserver/gen_qfsm?

Cheers
- Tilman

**Comment by muxspace** [comment]
Tilman,

Correct, there is nothing in bunny_farm to explicitly acknowledge messages. Shouldn't be hard to build that in or just acknowledge from your gen_qserver/gen_qfsm behavior.

Brian

**Comment by ypaq** [comment]
In order to enable `{no_ack, false}` bunny_farm needs some changes. `bunny_farm:consume` will always use `{no_ack, true}` (see https://github.com/muxspace/bunny_farm/blob/master/src/bunny_farm.erl#L80). You need to allow to subscribe accordingly. Just ack from somewhere else won't cut it.

**Comment by muxspace** [comment]
Good catch. It's still an easy change. You're welcome to submit a pull request if you implement this.

**Comment by ypaq** [comment]
Will see if I find the time. I decided to go with gen_bunny because of its monitoring. I like your clean code and the exhaustive tests though. Good job on your bunny_farm!
